Services/source/adcs/adcs_service.c

```diff
@@ -0,0 +1,1429 @@
+        uint8_t control;
+        status = HAL_ADCS_get_power_control(&control);
+        if (sizeof(control) + 1 > csp_buffer_data_size()) {
+            return_state = SATR_ERROR;
+        }
+        control = csp_hton32((uint32_t)control);
+        
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));  
+        memcpy(&packet->data[OUT_DATA_BYTE], &control, sizeof(control)); 
+        set_packet_length(packet, sizeof(control) + sizeof(int8_t) + 1);
+        break;
+    }
+        
+    case ADCS_SET_ATTITUDE_ANGLE: {
+        xyz angle;
+        memcpy(&angle, packet->data[IN_DATA_BYTE], sizeof(xyz));
+        status = ADCS_set_attitude_angle(angle);
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));  
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        break;
+    }
+
+    case ADCS_GET_ATTITUDE_ANGLE: {
+        xyz angle;
+        status = HAL_ADCS_get_attitude_angle(&angle);
+        if (sizeof(angle) + 1 > csp_buffer_data_size()) {
+            return_state = SATR_ERROR;
+        }
+        
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));  
+        memcpy(&packet->data[OUT_DATA_BYTE], &angle, sizeof(angle)); 
+        set_packet_length(packet, sizeof(angle) + sizeof(int8_t) + 1);
+        break;
+    }
+
+    case ADCS_SET_TRACK_CONTROLLER: {
+        xyz target;
+        memcpy(&target, packet->data[IN_DATA_BYTE], sizeof(xyz));
+        status = ADCS_set_track_controller(target);
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));  
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        break;
+    }
+        
+    case ADCS_GET_TRACK_CONTROLLER: {
+        xyz target;
+        status = HAL_ADCS_get_track_controller(&target);
+        if (sizeof(target) + 1 > csp_buffer_data_size()) {
+            return_state = SATR_ERROR;
+        }
+        
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));  
+        memcpy(&packet->data[OUT_DATA_BYTE], &target, sizeof(target)); 
+        set_packet_length(packet, sizeof(target) + sizeof(int8_t) + 1);
+        break;
+    }
+        
+    case ADCS_SET_LOG_CONFIG: {
+        uint8_t flags_arr = packet->data[IN_DATA_BYTE];
+        uint16_t period;
+        cnv8_16(&packet->data[IN_DATA_BYTE + 1], &period);
+        period = csp_ntoh16(period);
+        uint8_t dest = packet->data[IN_DATA_BYTE + 2];
+        uint8_t log = packet->data[IN_DATA_BYTE + 3];
+
+        status = HAL_ADCS_set_log_config(&flags_arr, period, dest, log);
+
+        if (sizeof(flags_arr) + 1 > csp_buffer_data_size()) {
+            return_state = SATR_ERROR;
+        }
+
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));  
+        memcpy(&packet->data[OUT_DATA_BYTE], &flags_arr, sizeof(flags_arr)); 
+        set_packet_length(packet, sizeof(flags_arr) + sizeof(int8_t) + 1);
+
+        break;
+    }
+        
+    case ADCS_GET_LOG_CONFIG: {
+        uint8_t flags_arr;
```

Comment:
This should be an array. Check the manual and/or handler code to find the proper size.
Similar comment for other flags_arr arrays or ones with other names.

Reply:
Sure I will get these fixed